### PR TITLE
enhance: development flow

### PR DESCRIPTION
**Pull request by lucasrodes**
This PR adds some development tools to ease the development and maintenance of this library (coverage reports, documentation, versioning, etc.). Find each of these new functionalities described in the sections below.

## Improve overall unit testing coverage
Added some additional tests. Now we are at 84% coverage (before that we were at 71%). We should set a threshold, so we ensure that we are always above that.

## Improve code style
Minor styling issues raised by flake8, mainly docstrings and other minor coding lines.

## Documentation
I created folder `docs` which will contain all the documentation config and output files. I have used [Sphinx](https://www.sphinx-doc.org/en/master/), which has been added to the dev dependencies. 

I also have set up a readthedocs.org page for the documentation: https://docs.owid.io/projects/datautils. Currently only a the page for the branch `feature/development` is operative.

The idea is that all the docstrings in the source code are rendered in a human-friendly format. Additional pages can be created using reST or markdown. The documentation at https://docs.owid.io/projects/datautils is updated with every commit on the branches that we choose (e.g. `main`), and also tagged versions that we select.

Note that this is in a very early stage. Also, the fact that we use `poetry` and do not have a `requirements.txt` file makes the readthedocs.org configuration a bit more tricky (https://github.com/readthedocs/readthedocs.org/issues/4912).

## Versioning
The version of the package appears in many files. To this end, I have added [bumpversion](https://github.com/c4urself/bump2version) as a dev dependency. It eases the process of maintaining all these annotations in sync.

I incorporated a command to the `Makefile`, so you can now run

```
make bump [part]
```

where `part` can be "major", "minor" or patch (based on [SemVer 2.0.0](https://semver.org/)).

## Reports
You can now generate reports that provide an overview of the code style of the library and the coverage of the unit tests:

```
make report
```

The upper command generates code style and testing coverage reports at `./reports/`.

## Coverage tests
Set up a space in Codecov to host our coverage reports. Added a step to the CI/CD pipeline to upload the coverage to Codecov.

Bow, for PR we should get a report on how well we are doing in terms of coverage; especially compared to the base. If the test coverage of the PR is lower than the base, PR will be rejected (there is some tolerance).

In the future, it'd be nice to add this coverage check to the pre-commit config, if possible.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,4 +1,6 @@
 [flake8]
 # Ignore some errors, since we autoformat them away already wherever possible
 # from https://github.com/psf/black/blob/main/.flake8
-ignore = E203, E266, E501, W503
+# ignore = E203, E266, E501, W503
+ignore = D107, W503, E203  # should align black and flake8 'rules'
+max-line-length = 120
```
Comment (pabloarosado) [inline .flake8:6]: Thanks for noting this, I dislike black so much for imposing by default such short lines that make the code unnecessarily long! Once I tried changing this parameter in the etl and it did not work, I hope here it works.

**Review by pabloarosado — APPROVED** [review]
Thanks for starting this! I'd appreciate having a better understanding of how to do documentation from now on, it's not clear to me what the workflow should be.

**Comment by lucasrodes** [comment]
@pabloarosado Thanks! I think once this is merged, we can have a meeting where I do some demo on how to use the various tools.

Once we consolidate this workflow, I think it makes sense to create a [template repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) so we can import this setup easily in other projects.

File: pyproject.toml [hunk 2]
```diff
@@ -19,19 +19,27 @@ structlog = "^21.5.0"
 colorama = "^0.4.4"
 mypy-boto3-s3 = "^1.21.23"
 owid-catalog = "^0.2.6"
-
+flake8-type-annotations = "^0.1.0"
```
Comment (Marigold) [inline pyproject.toml:22]: Shouldn't flake8-type-annotations be in `dev-dependencies`? Same for bumpversion?

**Review by Marigold — APPROVED** [review]
Love it!

File: pyproject.toml (hunk 2 above)
Comment (lucasrodes) [reply pyproject.toml]: Ups, yes - totally right!

**Pull request merged**